### PR TITLE
Prevent assertion of view with 0 width and or height

### PR DESCRIPTION
**Pull request by mackoj**
I don't think `swift-snapshot-testing` should assert when a view that it try to snapshot has 0 width, height or both. From my point of view it my be better to ignore it and create an artefact that represent it state.

A view can be in that state for a lot a justified reason(animation/state) and we might want to know that this view is actually with no width or height and it can change so being able to know that it is a 0 can actually be important.

Yes I could have used the `recursiveDescription` but it show way too many information so I made this PR.

This only support iOS I can do it for macOS too if needed.

### CODE REVIEW

File: Sources/SnapshotTesting/AssertSnapshot.swift [hunk 1]
```diff
@@ -240,6 +240,13 @@ public func verifySnapshot<Value, Format>(
       let data = try Data(contentsOf: snapshotFileUrl)
       let reference = snapshotting.diffing.fromData(data)
 
+      #if os(iOS)
```
Comment (stephencelis) [inline Sources/SnapshotTesting/AssertSnapshot.swift]:
```suggestion
      #if os(iOS) || os(tvOS)
```

File: Sources/SnapshotTesting/Common/View.swift [hunk 2]
```diff
@@ -628,6 +628,14 @@ extension View {
     #endif
     return nil
   }
+  #if os(iOS)
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Common/View.swift]:
```suggestion
  #if os(iOS) || os(tvOS)
```

File: Tests/SnapshotTestingTests/SnapshotTestingTests.swift [hunk 3]
```diff
@@ -704,6 +707,25 @@ final class SnapshotTestingTests: XCTestCase {
     }
     #endif
   }
+  
+  func testViewWithZeroHeightOrWidth() {
+    #if os(iOS)
```
Comment (stephencelis) [inline Tests/SnapshotTestingTests/SnapshotTestingTests.swift]:
```suggestion
    #if os(iOS) || os(tvOS)
```

**Comment by mackoj** [comment]
Thanks for the review @stephencelis I have done the modifications you suggested.

**Comment by mackoj** [comment]
@stephencelis I know stewardship on open source project can be time consuming  be if you could find some time to merge this it would be great thanks. 🙏

File: Sources/SnapshotTesting/Snapshotting/UIImage.swift [hunk 4]
```diff
@@ -32,6 +32,16 @@ extension Diffing where Value == UIImage {
       )
     }
   }
+  
+  
+  /// Used when the image size has no width or no height to generated the default empty image
+  private static func emptyImage() -> UIImage {
+    let label = UILabel(frame: CGRect(x: 0, y: 0, width: 200, height: 30))
+    label.backgroundColor = .red
+    label.text = "NO IMAGE GENERATED"
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Snapshotting/UIImage.swift:41]: @mackoj What do you think of enhancing this error message with more information? Something like: 
```suggestion
    label.text = "Error: No image could be generated for this view as its size was zero. Please set an explicit size in the test."
```
If you don't have time to make the adjustments, that's fine! We can merge as-is and try to make the changes ourselves soon.

**Comment by mackoj** [comment]
@stephencelis it's ready for re-review I have updated the text as you ask and regenerate fixture and update the label for the new text to fit in.

**Comment by stephencelis** [comment]
Looks great! Thanks for taking the time 😄

**Pull request merged**